sdk/digitaltwins/azure-digitaltwins-core/API design.md

```diff
@@ -0,0 +1,557 @@
+# Digital Twin API Design Doc
+
+## Async APIs
+This library utilizes [Project Reactor](https://projectreactor.io/) to provide consumers with async APIs.
+
+Note:
+- [Mono](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html) is a reactive streams publisher which completes an async operation after emiting a single element, or an error.
+- [Flux](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html) is a reactive streams publisher which completes an async operation after emiting 0 to N elements, or an error.
+
+This document outlines the APIs for the Digital Twin SDK
+
+## Azure.Core usage
+Within this SDK, we will make use of several Azure.Core library classes:
+
+Pagination:
+
+[PagedIterable\<T>](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java): For synchronous APIs - This class provides utility to iterate over PagedResponse using Stream and Iterable interfaces.
+
+Code sample using `Stream` by page:
```

Comment:
From java docs here: https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/samples/java/com/azure/core/http/rest/PagedIterableJavaDocCodeSnippets.java